src/zenml/cli/stack.py

```diff
@@ -174,6 +174,267 @@ def register_stack(
+    is_flag=True,
+    required=False,
+)
+@click.option(
+    "-f",
+    "--feature_store",
+    "feature_store_flag",
+    help="Name of the feature store to remove from this stack.",
+    is_flag=True,
+    required=False,
+)
+def remove_stack_component(
+    stack_name: str,
+    container_registry_flag: Optional[bool] = False,
+    step_operator_flag: Optional[bool] = False,
+    secrets_manager_flag: Optional[bool] = False,
+    feature_store_flag: Optional[bool] = False,
+) -> None:
+    """Remove stack components from a stack."""
+    with console.status(f"Updating stack `{stack_name}`...\n"):
+        repo = Repository()
+        try:
+            current_stack = repo.get_stack(stack_name)
+        except KeyError:
+            cli_utils.error(
+                f"Stack `{stack_name}` cannot be updated as it does not exist.",
+            )
+        stack_components = current_stack.components
+
+        if container_registry_flag:
+            stack_components = {
+                key: value
+                for (key, value) in stack_components.items()
+                if key != StackComponentType.CONTAINER_REGISTRY
+            }
+
+        if step_operator_flag:
+            stack_components = {
+                key: value
+                for (key, value) in stack_components.items()
+                if key != StackComponentType.STEP_OPERATOR
+            }
+
+        if secrets_manager_flag:
+            stack_components = {
+                key: value
+                for (key, value) in stack_components.items()
+                if key != StackComponentType.SECRETS_MANAGER
+            }
+
+        if feature_store_flag:
+            stack_components = {
+                key: value
+                for (key, value) in stack_components.items()
+                if key != StackComponentType.FEATURE_STORE
+            }
+
+        stack_ = Stack.from_components(
+            name=stack_name, components=stack_components
+        )
+        repo.update_stack(stack_)
+        cli_utils.declare(f"Stack `{stack_name}` successfully updated!")
+
+
+@stack.command("rename")
+@click.argument("current_stack_name", type=str, required=True)
+@click.argument("new_stack_name", type=str, required=True)
+def rename_stack(
+    current_stack_name: str,
+    new_stack_name: str,
+) -> None:
+    """Rename a stack."""
+    with console.status(f"Renaming stack `{current_stack_name}`...\n"):
+        repo = Repository()
+        try:
+            current_stack = repo.get_stack(current_stack_name)
+        except KeyError:
+            cli_utils.error(
+                f"Stack `{current_stack_name}` cannot be renamed as it does not exist.",
+            )
+        stack_components = current_stack.components
+
+        new_stack_ = Stack.from_components(
+            name=new_stack_name, components=stack_components
+        )
+        try:
+            repo.deregister_stack(current_stack_name)
+            repo.register_stack(new_stack_)
```

Comment:
This (effectively) uses a de-register / re-register to simulate a stack rename. While functionally correct, it would be much better if we had some sort of `rename_stack` method in the base stack store API that the various stack stores would need to implement. Conceptually speaking, a rename is much less intrusive than a delete/recreate.

However, I do understand the convenience of doing delete/create instead of having to implement an entirely new method for 3 different store types (yaml, sql and rest) :) . Fine by me if you just want to write this off as a TODO.

Reply:
Yep. I'll see how far I get today with things. I think when I was writing the stack rename I hadn't fully appreciated how I'd have to do all of that. It might end up being nicer to have a `rename_stack` method instead, though that does require a lot more work (and all the corresponding tests). I'll see!

Reply:
Fixed this.